rest-service/manager_rest/resource_manager.py

```diff
@@ -1339,6 +1370,54 @@ def add_node_instance_from_dict(self, instance_dict):
         instance_dict['resource_availability'] = resource_availability
         instance_dict['private_resource'] = private_resource
 
+    def create_operation(self, operation_id, name, dependencies,
+                         parameters, type, graph_id=None,
+                         graph_storage_id=None):
+        # allow passing graph_storage_id directly as an optimization, so that
+        # we don't need to query for the graph based on display id
+        if (graph_id and graph_storage_id) or \
+                (not graph_id and not graph_storage_id):
```

Comment:
Can't we use xor (^?) here?

Reply:
(well- not (graph_id ^ graph_storage_id))

Reply:
i think this is more readable, because the xor would require bool() conversions, because if the ids are ints, then xor might well return zero even if both are passed (if they're equal)